securitycenter/docs/snippets_findings.py

```diff
@@ -0,0 +1,603 @@
+    # created previously (you can use list_sources to find a specific one).
+    # Its format is:
+    # source_name = "organizations/{organization_id}/sources/{source_id}"
+    # e.g.:
+    # source_name = "organizations/111122222444/sources/1234"
+    # Get the old policy so we can do an incremental update.
+    old_policy = client.get_iam_policy(source_name)
+    print("Old Policy: {}".format(old_policy))
+
+    # Setup a new IAM binding.
+    binding = policy_pb2.Binding()
+    binding.role = "roles/securitycenter.findingsEditor"
+    # user_email is an e-mail address known to Cloud IAM (e.g. a gmail address).
+    # user_mail = user@somedomain.com
+    binding.members.append("user:{}".format(user_email))
+
+    # Setting the e-tag avoids over-write existing policy
+    updated = client.set_iam_policy(
+        source_name, {"etag": old_policy.etag, "bindings": [binding]}
+    )
+
+    print("Updated Policy: {}".format(updated))
+
+    # [END update_source_iam]
+    assert any(
+        member == "user:csccclienttest@gmail.com"
+        for member in chain.from_iterable(
+            binding.members for binding in updated.bindings
+        )
+    )
+
+
+def test_list_source(organization_id):
+    """Lists finding sources."""
+    i = -1
+    # [START list_sources]
+    from google.cloud import securitycenter as securitycenter
+
+    # Create a new client.
+    client = securitycenter.SecurityCenterClient()
+    # organization_id is the numeric ID of the organization. e.g.:
+    # organization_id = "111122222444"
+    org_name = "organizations/{org_id}".format(org_id=organization_id)
+
+    # Call the API and print out each existing source.
+    for i, source in enumerate(client.list_sources(org_name)):
+        print(i, source)
+    # [END list_sources]
+    assert i >= 0
+
+
+def test_create_finding(source_name):
+    """Creates a new finding."""
+    # [START create_finding]
+    from google.cloud import securitycenter as securitycenter
+    from google.cloud.securitycenter_v1.proto.finding_pb2 import Finding
+    from google.protobuf.timestamp_pb2 import Timestamp
+
+    # Create a new client.
+    client = securitycenter.SecurityCenterClient()
+
+    # Use the current time as the finding "event time".
+    now_proto = Timestamp()
+    now_proto.GetCurrentTime()
+
+    # source_name is the resource path for a source that has been
+    # created previously (you can use list_sources to find a specific one).
+    # Its format is:
+    # source_name = "organizations/{organization_id}/sources/{source_id}"
+    # e.g.:
+    # source_name = "organizations/111122222444/sources/1234"
+
+    # Controlled by caller.
+    finding_id = "samplefindingid"
+
+    # The resource this finding applies to.  The CSCC UI can link
+    # the findings for a resource to the corresponding Asset of a resource
+    # if there are matches.
+    resource_name = "//cloudresourcemanager.googleapis.com/organizations/11232"
```

Comment:
Where does this come from?

Reply:
Arbitrary name (it doesn't need to exist).